### ISSUE TITLE
API Documentation Audit Quest Issue

### ISSUE THREAD

**Issue by bmac** [emberjs/data#4394]
The Ember Data team would like to improve the api docs by auditing every file to ensure the documentation is correct and the public apis all have examples showing their use. If you would like to help out with this issue claim a file from the list below by commenting on this issue. The file is yours for 24 hours, let us know if you need more time. Please read thought the file you have claimed to ensure all of the API doc blocks meet the following requirements: 
- All documentation should be verified as correct
- When it makes sense, there should be a premable blocks for the class describing the major parts of the framework and their use [Example](https://github.com/emberjs/data/blob/4fb5bb4727db65680de97f870ce05469cf99d0e5/addon/transform.js#L3-L39)
- All public methods should have a description and example usage. [Example](https://github.com/emberjs/data/blob/4fb5bb4727db65680de97f870ce05469cf99d0e5/addon/transform.js#L41-L57)
- All methods have `@param` and `@return` annotations with the type. [Example](https://github.com/emberjs/data/blob/4fb5bb4727db65680de97f870ce05469cf99d0e5/addon/adapter.js#L116-L121)
- If you know a method or a class has been added since 1.13 it should have `@since` tag with the version where it was added. [Example](https://github.com/emberjs/ember.js/blob/ed298ced885b2f4b4435fbccb26b24b9a24bb68f/packages/ember-runtime/lib/system/service.js#L26)
- Private methods can be left undocumented

List of Ember Data files:
- [ ] `addon/store.js`
- [ ] `addon/setup-container.js`
- [ ] `addon/serializer.js`
- [ ] `addon/relationships.js`
- [ ] `addon/model.js`
- [ ] `addon/transform.js`
- [ ] `addon/index.js`
- [ ] `addon/attr.js`
- [ ] `addon/adapter.js`
- [ ] `addon/-private`
- [ ] `addon/-private/utils.js`
- [ ] `addon/-private/transforms.js`
- [ ] `addon/-private/serializers.js`
- [ ] `addon/-private/features.js`
- [ ] `addon/-private/debug.js`
- [ ] `addon/-private/core.js`
- [ ] `addon/-private/adapters.js`
- [ ] `addon/-private/utils/parse-response-headers.js`
- [ ] `addon/-private/instance-initializers/initialize-store-service.js`
- [ ] `addon/-private/initializers/transforms.js`
- [ ] `addon/-private/initializers/store.js`
- [ ] `addon/-private/initializers/store-injections.js`
- [ ] `addon/-private/initializers/data-adapter.js`
- [ ] `addon/-private/ext/date.js`
- [ ] `addon/-private/adapters/build-url-mixin.js`
- [ ] `addon/-private/transforms/string.js`
- [ ] `addon/-private/transforms/number.js`
- [ ] `addon/-private/transforms/date.js`
- [ ] `addon/-private/transforms/boolean.js`
- [ ] `addon/-private/system/relationship-meta.js`
- [ ] `addon/-private/system/references.js`
- [ ] `addon/-private/system/record-arrays.js`
- [ ] `addon/-private/system/record-array-manager.js`
- [ ] `addon/-private/system/promise-proxies.js`
- [ ] `addon/-private/system/normalize-model-name.js`
- [ ] `addon/-private/system/normalize-link.js`
- [ ] `addon/-private/system/model.js`
- [ ] `addon/-private/system/is-array-like.js`
- [ ] `addon/-private/system/empty-object.js`
- [ ] `addon/-private/system/debug.js`
- [ ] `addon/-private/system/container-proxy.js`
- [ ] `addon/-private/system/clone-null.js`
- [ ] `addon/-private/system/many-array.js`
- [ ] `addon/-private/system/store.js`
- [ ] `addon/-private/system/snapshot.js`
- [ ] `addon/-private/system/snapshot-record-array.js`
- [ ] `addon/-private/system/ordered-set.js`
- [ ] `addon/-private/system/coerce-id.js`
- [ ] `addon/-private/system/debug/debug-info.js`
- [ ] `addon/-private/system/debug/debug-adapter.js`
- [ ] `addon/-private/system/references/reference.js`
- [ ] `addon/-private/system/references/record.js`
- [ ] `addon/-private/system/references/has-many.js`
- [ ] `addon/-private/system/references/belongs-to.js`
- [ ] `addon/-private/system/store/serializers.js`
- [ ] `addon/-private/system/store/container-instance-cache.js`
- [ ] `addon/-private/system/store/common.js`
- [ ] `addon/-private/system/store/serializer-response.js`
- [ ] `addon/-private/system/store/finders.js`
- [ ] `addon/-private/system/relationships/has-many.js`
- [ ] `addon/-private/system/relationships/belongs-to.js`
- [ ] `addon/-private/system/relationships/ext.js`
- [ ] `addon/-private/system/relationships/state/relationship.js`
- [ ] `addon/-private/system/relationships/state/has-many.js`
- [ ] `addon/-private/system/relationships/state/belongs-to.js`
- [ ] `addon/-private/system/relationships/state/create.js`
- [ ] `addon/-private/system/record-arrays/filtered-record-array.js`
- [ ] `addon/-private/system/record-arrays/record-array.js`
- [ ] `addon/-private/system/record-arrays/adapter-populated-record-array.js`
- [ ] `addon/-private/system/model/errors.js`
- [ ] `addon/-private/system/model/attr.js`
- [ ] `addon/-private/system/model/states.js`
- [ ] `addon/-private/system/model/model.js`
- [ ] `addon/-private/system/model/internal-model.js`
- [ ] `addon/adapters/json-api.js`
- [ ] `addon/adapters/errors.js`
- [ ] `addon/adapters/rest.js`
- [ ] `addon/serializers/embedded-records-mixin.js`
- [ ] `addon/serializers/rest.js`
- [ ] `addon/serializers/json.js`
- [ ] `addon/serializers/json-api.js`

Thank you for your help. Don't hesitate to ask if you have any additional questions.

**Comment by zachgarwood** [comment]
I'll start with `addon/serializer.js`

**Comment by courajs** [comment]
Do you have anything in mind for the import and re-export files? `addon/store.js`, `addon/model.js` etc.

**Comment by bmac** [comment]
Nope. For those files you can just comment in this thread and say you have verified them and they do not need any documentation changes.

**Comment by courajs** [comment]
The following files are only import/re-export files, so should need no documentation changes:
- `addon/setup-container.js `
- `addon/relationships.js`
- `addon/index.js`

**Comment by bmac** [comment]
Thanks @courajs

**Comment by acorncom** [comment]
Another item I've just noticed is that transform options that were implemented in 2.5 haven't been documented anywhere other than the blog post announcing it (http://emberjs.com/blog/2016/05/03/ember-data-2-5-released.html#toc_code-ds-transform-pass-options-code).  Happy to open a separate issue if we'd rather have that handled there ...
